pkg/virt-launcher/virtwrap/api/converter.go

```diff
@@ -460,6 +461,61 @@ func Convert_v1_VirtualMachine_To_api_Domain(vmi *v1.VirtualMachineInstance, dom
 		volumes[volume.Name] = volume.DeepCopy()
 	}
 
+	dedicatedThreads := 0
+	autoThreads := 0
+	useIOThreads := false
+	threadPoolLimit := 1
+
+	if vmi.Spec.Domain.IOThreadsPolicy != nil {
+		useIOThreads = true
+
+		if (*vmi.Spec.Domain.IOThreadsPolicy) == "auto" {
+			numCPUs := 1
+			// Requested CPU's is guaranteed to be no greater than the limit
+			if cpuRequests, ok := vmi.Spec.Domain.Resources.Requests[k8sv1.ResourceCPU]; ok {
+				numCPUs = int(cpuRequests.Value())
+			} else if cpuLimit, ok := vmi.Spec.Domain.Resources.Limits[k8sv1.ResourceCPU]; ok {
+				numCPUs = int(cpuLimit.Value())
+			}
+
+			threadPoolLimit = numCPUs * 2
+
+			if vmi.Spec.Domain.CPU != nil {
+				threadPoolLimit = 2 * int(vmi.Spec.Domain.CPU.Cores)
```

Comment:
@stu-gott I don't think we need this at all. Without the cpu pinning in the picture, vmi.Spec.Domain.CPU.Cores only affect the guest topology and it shouldn't affect the calculation of vcpus.